packages/@aws-cdk/aws-sagemaker/lib/endpoint-config.ts

```diff
@@ -0,0 +1,313 @@
+      public endpointConfigArn = endpointConfigArn;
+      public endpointConfigName = endpointConfigName;
+    }
+
+    return new Import(scope, id, {
+      environmentFromArn: endpointConfigArn,
+    });
+  }
+
+  /**
+   * Imports an EndpointConfig defined either outside the CDK or in a different CDK stack.
+   * @param scope the Construct scope.
+   * @param id the resource id.
+   * @param endpointConfigName the name of the endpoint configuration.
+   */
+  public static fromEndpointConfigName(scope: Construct, id: string, endpointConfigName: string): IEndpointConfig {
+    const endpointConfigArn = cdk.Stack.of(scope).formatArn({
+      service: 'sagemaker',
+      resource: 'endpoint-config',
+      resourceName: endpointConfigName,
+    });
+
+    class Import extends cdk.Resource implements IEndpointConfig {
+      public endpointConfigArn = endpointConfigArn;
+      public endpointConfigName = endpointConfigName;
+    }
+
+    return new Import(scope, id, {
+      environmentFromArn: endpointConfigArn,
+    });
+  }
+
+  /**
+   * The ARN of the endpoint configuration.
+   */
+  public readonly endpointConfigArn: string;
+  /**
+   * The name of the endpoint configuration.
+   */
+  public readonly endpointConfigName: string;
+
+  private readonly _instanceProductionVariants: { [key: string]: InstanceProductionVariant; } = {};
+
+  constructor(scope: Construct, id: string, props: EndpointConfigProps = {}) {
+    super(scope, id, {
+      physicalName: props.endpointConfigName,
+    });
+
+    (props.instanceProductionVariants || []).map(p => this.addInstanceProductionVariant(p));
+
+    // create the endpoint configuration resource
+    const endpointConfig = new CfnEndpointConfig(this, 'EndpointConfig', {
+      kmsKeyId: (props.encryptionKey) ? props.encryptionKey.keyArn : undefined,
+      endpointConfigName: this.physicalName,
+      productionVariants: cdk.Lazy.any({ produce: () => this.renderInstanceProductionVariants() }),
+    });
+    this.endpointConfigName = this.getResourceNameAttribute(endpointConfig.attrEndpointConfigName);
+    this.endpointConfigArn = this.getResourceArnAttribute(endpointConfig.ref, {
+      service: 'sagemaker',
+      resource: 'endpoint-config',
+      resourceName: this.physicalName,
+    });
+  }
+
+  /**
+   * Add production variant to the endpoint configuration.
+   *
+   * @param props The properties of a production variant to add.
+   */
+  public addInstanceProductionVariant(props: InstanceProductionVariantProps): void {
+    if (props.variantName in this._instanceProductionVariants) {
+      throw new Error(`There is already a Production Variant with name '${props.variantName}'`);
+    }
+    this.validateProps(props);
+    this._instanceProductionVariants[props.variantName] = {
+      acceleratorType: props.acceleratorType,
+      initialInstanceCount: props.initialInstanceCount || 1,
+      initialVariantWeight: props.initialVariantWeight || 1.0,
```

Comment:
I don't think TypeScript values the decimal, so `initialVariantWeight: 1.0` will show up in the template as `initialVariantWeight: 1`. I checked the template generated by the integ test, and this is true. Not sure if it's a real problem, but calling it out.

Reply:
Nice catch! Fortunately, it doesn't look like CloudFormation nor SageMaker minds. I re-ran the integration test with `--no-clean` and then queried the resource via the AWS CLI:
```
aws sagemaker describe-endpoint-config --endpoint-config-name EndpointConfigFD7B6F91-BOORCu7hWABi
{
    "EndpointConfigName": "EndpointConfigFD7B6F91-BOORCu7hWABi",
    "EndpointConfigArn": "arn:aws:sagemaker:us-east-1:231206303950:endpoint-config/endpointconfigfd7b6f91-boorcu7hwabi",
    "ProductionVariants": [
        {
            "VariantName": "firstVariant",
            "ModelName": "ModelWithArtifactAndVpcModel30604F15-P7JnQBqaoYRy",
            "InitialInstanceCount": 1,
            "InstanceType": "ml.m5.large",
            "InitialVariantWeight": 1.0
        },
        {
            "VariantName": "secondVariant",
            "ModelName": "ModelWithArtifactAndVpcModel30604F15-P7JnQBqaoYRy",
            "InitialInstanceCount": 1,
            "InstanceType": "ml.t2.medium",
            "InitialVariantWeight": 1.0
        },
        {
            "VariantName": "thirdVariant",
            "ModelName": "ModelWithoutArtifactAndVpcModel9A8AD144-ZDL4AYJLv4G4",
            "InitialInstanceCount": 1,
            "InstanceType": "ml.t2.medium",
            "InitialVariantWeight": 2.0
        }
    ],
    "CreationTime": "2022-11-10T14:31:45.936000-08:00"
}
```

As you can see, the variant weights are still being treated as floating point values even though they are expressed as integers in the template.